### ISSUE TITLE
MinniZinc 2.6.3: flattener error (wrong answer)

### ISSUE THREAD

**Issue by matsc-at-sics-se** [MiniZinc/libminizinc#620]
This is a continuation of #565 , which apparently was not quite fixed.  This model is infeasible:
```
var bool: c;
var bool: d;
constraint
    not(not(true xor c) -> ((d -> c) xor false));

output [show([c,d]) ++ "\n"];
```
but MiniZinc finds two wrong answers:
```
$ minizinc -a bar.mzn
[true, false]
----------
[true, true]
----------
==========
```

**Comment by Dekker1** [comment]
It appears that this issue has already been resolved on the current `develop` branch. I've added an additional regression test in 4b297c93254c86337c019fd47d8be341b96e600a to ensure the issue doesn't reappear.

**Comment by cyderize** [comment]
This should now be fixed in version 2.7.0
